src/mds/MDSMap.h

```diff
@@ -634,6 +658,8 @@ class MDSMap {
 
   mds_rank_t tableserver = 0;   // which MDS has snaptable
   mds_rank_t root = 0;          // which MDS has root directory
+  std::unordered_set<mds_gid_t> qdb_cluster_members;
+  mds_gid_t qdb_cluster_leader = MDS_GID_NONE;
```

Comment:
We learned that we _could_ replicate with standbys, and the only problem was some other logic which I didn't have time to deal with, the one that marked our standby as impostor. Otherwise, this works very well, and I'm going to have a separate commit that addresses the impostor problem to have our DB held by the standby-replays.

> departure from the current structure from the MDSMap for no functional reason

There is a great reason for it, which is to utilize available daemons for useful work. GIDs are all over the map, listing the daemons that are part of the ceph cluster. I fail to see how this approach could be "breaking"

Reply:
> Also, the MDS are still independently computing the leader which will allow for divergence in behavior.

They shouldn't be, have I missed something?

Reply:
> > Also, the MDS are still independently computing the leader which will allow for divergence in behavior.
> 
> They shouldn't be, have I missed something?
> 
> See `MDSRank::quiesce_cluster_update()`:
> 
> ```
>     mdsmap->get_quiesce_db_cluster(membership.leader, membership.members);
> ```

I see, thanks.

Reply:
Following a talk a TODO here is to add a check to the sanity function to validate the consistency of this field